### ISSUE TITLE
Download through company proxy

### ISSUE THREAD

**Issue by thisispax** [winfinit/mongodb-download#11]
We are using a company-proxy to access the internet. Unfortunately the request doesn't handle this.

Can you please add proxy-support by just adding following:

``` javascript
var HttpsProxyAgent = require('https-proxy-agent');
...
var options = {
            url: DOWNLOAD_URI;
        };

        if(process.env.HTTPS_PROXY) {
            debug("setting proxy for request");
            options['agent'] = new HttpsProxyAgent(process.env.HTTPS_PROXY);
        }

        var request = http.get(options, function(response) {
...
```

**Comment by thisispax** [comment]
see https://github.com/winfinit/mongodb-download/pull/12

**Comment by winfinit** [comment]
@pax1878 I've actually did include already an ability to pass agent as part of the options via http_opts object. I didn't follow your suggestion, because i don't like the idea of including extra proxy module for general public. I didn't reply to this issue yet because i want to propagate this functionality to prebuilt and mockgoose modules.

**Comment by thisispax** [comment]
Ok. But why should prebuilt or mockgoose care about a download which happens in this app? Acutally I can live with your idea. But we need a solution soon.

Btw. great job ;-)

**Comment by winfinit** [comment]
@pax1878 oh you are using it as a standalone module, then that is already in 1.3.0 that is on npmjs.

``` javascript
var HttpsProxyAgent = require('https-proxy-agent');
var download = require('mongoldb-download');

download({
  http_opts: {
     agent: new HttpsProxyAgent("http://myproxy.example.com")
  } // extra options that one would want to pass to http request
}, function (err, location) {
  // location will be the path of the archive that it downloaded.
})
```

just noticed, there is a typo in a doc, so ill fix that, but that won't affect above snippet.

**Comment by winfinit** [comment]
@pax1878 just published mockgoose 6.0.0, there is a breaking API change, however this would produce much more reliable results, and it also includes dependencies that will let you do what you want.

you can now specify https_proxy via npm install flag

```
npm install --https-proxy="https://example.com"
```

or by setting an environment variable 

```
#*nix
export https_proxy="https://example.com"
#win32
set https_proxy="https://example.com"
```

i wants able to test this portion yet, so your results would be greatly appreciated.

**Comment by thisispax** [comment]
thanks!
